From 2276e777464ee6a65fe1bf050598554d6e73edfc Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 08 Apr 2019 09:58:31 +0300
Subject: [PATCH] [SW-0] preboot: remove u-boot headers and macros

Remove more u-boot headers as well as search directories.
Also remove unused GCC flags.

TODO: we are left with u-boot header linux/io.h which needed
by SPI driver and should be replaced with some alternative implementation

Change-Id: I0299e587d2811681e9b7f8d14b13c1377dede947
---

diff --git a/Makefile b/Makefile
index a7f8ef4..dcd3cd4 100644
--- a/Makefile
+++ b/Makefile
@@ -2,15 +2,11 @@
 OBJCOPY = $(CROSS_COMPILE)objcopy
 CFLAGS += -I$(dir $(BR2_EXTERNAL))/specs/$(HL_DEV)/asic_reg
 CFLAGS += -I$(dir $(BR2_EXTERNAL))/embedded/specs
-CFLAGS += -g -nostdinc -fno-pic -fno-builtin
-CFLAGS += -D__KERNEL__ -D__UBOOT__ -D__ARM__ -march=armv8-a -D__LINUX_ARM_ARCH__=8
-CFLAGS += -isystem ../../host/opt/ext-toolchain/lib/gcc/aarch64-linux-gnu/6.4.1/include
+CFLAGS += -D__KERNEL__
 CFLAGS += -I ../uboot-custom/include
 CFLAGS += -I ../uboot-custom/arch/arm/include
-CFLAGS += -I ../uboot-custom/arch/arm/mach-habanalabs/include
-CFLAGS += -include ../uboot-custom/include/linux/kconfig.h
 CFLAGS += -Wall -Werror
-CFLAGS += -c
+CFLAGS += -g -c
 CFLAGS += -I ./ -I ./common/
 
 ifeq ($(HL_TARGET),pldm)
diff --git a/common/spi.c b/common/spi.c
index e72020a..532d8a8 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -5,9 +5,7 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <config.h>
 #include <linux/io.h>
-#include <linux/sizes.h>
 
 #include <psoc_gpio3_regs.h>
 #include <psoc_spi_regs.h>
diff --git a/common/spi.h b/common/spi.h
index 7e39071..722c2e9 100644
--- a/common/spi.h
+++ b/common/spi.h
@@ -8,7 +8,6 @@
 #ifndef _HL_SPI_H_
 #define _HL_SPI_H_
 
-
 void spi_copy_spl(unsigned int spl_offset);
 u64 get_cs_pad_sel(void);
 
diff --git a/common/uart.c b/common/uart.c
index 4cafc5f..88f2b9b 100644
--- a/common/uart.c
+++ b/common/uart.c
@@ -5,11 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <linux/io.h>
-
 #include <psoc_uart_0_regs.h>
 
 #include "preboot.h"
+#include "uart.h"
 
 #define UART_SR_TXFULL			(1 << 0) /* TX Buffer Full */
 #define STR_UINT_SZ			8
diff --git a/common/uart.h b/common/uart.h
index 0d298f6..4935d05 100644
--- a/common/uart.h
+++ b/common/uart.h
@@ -5,9 +5,10 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
+#include <stdbool.h>
+
 #ifndef _HL_UART_H_
 #define _HL_UART_H_
-
 
 void preboot_puts_raw(const char *str);
 void preboot_puts(const char *str);
diff --git a/goya/pll.c b/goya/pll.c
index 11b6a71..817934b 100644
--- a/goya/pll.c
+++ b/goya/pll.c
@@ -1,7 +1,4 @@
-#include <command.h>
-#include <linux/delay.h>
 #include <linux/io.h>
-#include <linux/types.h>
 
 #include "preboot.h"
 #include "uart.h"
